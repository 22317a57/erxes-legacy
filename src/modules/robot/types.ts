export type IFeature = {
  name: string;
  text: string;
};
