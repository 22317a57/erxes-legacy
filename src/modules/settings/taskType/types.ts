export interface ITaskType {
  _id: string;
  name: string;
  icon: string;
}

// query types

export type TaskTypeQueryResponse = {
  taskTypes: ITaskType[];
  loading: boolean;
  refetch: () => void;
};

export type MutationVariables = {
  _id?: string;
  name?: string;
  icon?: string;
};

// mutation types

export type AddMutationResponse = {
  addMutation: (mutation: { variables: MutationVariables }) => Promise<any>;
};

export type EditMutationResponse = {
  editMutation: (mutation: { variables: MutationVariables }) => Promise<any>;
};

export type RemoveMutationResponse = {
  removeMutation: (mutation: { variables: { _id: string } }) => Promise<any>;
};
