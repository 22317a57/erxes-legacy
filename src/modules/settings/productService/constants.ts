export const TYPES = {
  PRODUCT: 'product',
  SERVICE: 'service',
  ALL: ['product', 'service']
};

export const PRODUCT_INFO = {
  name: 'Name',
  description: 'Description',
  sku: 'Sku',

  ALL: [
    { field: 'name', label: 'Name' },
    { field: 'description', label: 'Description' },
    { field: 'sku', label: 'Sku' }
  ]
};

export const SERVICE_INFO = {
  name: 'Name',
  description: 'Description',
  sku: 'Sku',

  ALL: [
    { field: 'name', label: 'Name' },
    { field: 'description', label: 'Description' },
    { field: 'sku', label: 'Sku' }
  ]
};
