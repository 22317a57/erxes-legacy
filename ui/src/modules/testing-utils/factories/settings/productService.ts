import * as Factory from 'factory.ts';
import { IProduct, IProductCategory, IProductCategoryDoc, IProductDoc } from 'modules/settings/productService/types';
import { tagFactory } from '../tags';

export const productDocFactory = Factory.Sync.makeFactory<IProductDoc>({
  _id: '',
  type: '',
  name: '',
  description: '',
  sku: '',
  createdAt: new Date,
  customFieldsData: {}
});

export const productCategoryDocFactory = Factory.Sync.makeFactory<IProductCategoryDoc>({
  _id: '1',
  name: 'nmma',
  description: 'desc',
  parentId: '2',
});

export const productCategoryFactory = Factory.Sync.makeFactory<IProductCategory>({
  _id: '1',
  name: 'nmma',
  order: '1',
  code: 'code',
  description: 'desc',
  parentId: '1',
  createdAt: new Date,
  productCount: 0,
  isRoot: false
});

export const productFactory = Factory.Sync.makeFactory<IProduct>({
  _id: '1',
  name: 'erxes',
  type: 'type',
  categoryId: '1',
  description: 'desc',
  getTags: [tagFactory.build()],
  sku: '0',
  code: '123',
  unitPrice: 0,
  customFieldsData: {},
  createdAt: new Date,
  category: productCategoryFactory.build()
});