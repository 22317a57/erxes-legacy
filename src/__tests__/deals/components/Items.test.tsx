// test passed
import { mount, shallow } from 'enzyme';
import Items from 'modules/deals/components/portable/Items';
import { IProduct } from 'modules/settings/productService/types';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

describe('Items component', () => {
  const testProducts: IProduct[] = [
    {
      _id: 'pro123',
      name: 'qwe',
      type: 's1',
      description: 'blabla',
      sku: 'sku1',
      createdAt: new Date()
    },
    {
      _id: 'pro23',
      name: 'qe',
      type: 's2',
      description: 'blablabla',
      sku: 'sku2',
      createdAt: new Date()
    }
  ];

  const defaultProps = {
    items: testProducts,
    color: 'green'
  };

  test('renders shallow successfully', () => {
    shallow(<Items {...defaultProps} />);
  });

  test('renders mount with default props', () => {
    const control = mount(<Items {...defaultProps} />);
    const props = control.props();

    expect(props).toMatchObject(defaultProps);
  });

  test('snapshot matches', () => {
    const rendered = renderer.create(<Items {...defaultProps} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
