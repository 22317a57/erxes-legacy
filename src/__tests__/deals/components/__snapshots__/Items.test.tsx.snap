// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items component snapshot matches 1`] = `
Array [
  <div>
    <span
      className="sc-kjoXOD cxmnoN"
      color="green"
    />
    qwe
  </div>,
  <div>
    <span
      className="sc-kjoXOD cxmnoN"
      color="green"
    />
    qe
  </div>,
]
`;
